File: src/2025-03-04-12-20-21--deepseek-vr-0304/go/exercises/practice/wordy/wordy.go
package wordy

import (
	"strconv"
	"strings"
)

func Answer(question string) (int, bool) {
	if !strings.HasPrefix(question, "What is ") || !strings.HasSuffix(question, "?") {
		return 0, false
	}
	
	expr := strings.TrimSpace(strings.TrimSuffix(strings.TrimPrefix(question, "What is "), "?"))
	if expr == "" {
		return 0, false
	}

	// Normalize operations to operator symbols
	expr = strings.ReplaceAll(expr, "multiplied by", "*")
	expr = strings.ReplaceAll(expr, "divided by", "/")
	expr = strings.ReplaceAll(expr, "plus", "+")
	expr = strings.ReplaceAll(expr, "minus", "-")

	tokens := strings.Fields(expr)
	if len(tokens) == 0 {
		return 0, false
	}

	// Parse first number
	result, err := strconv.Atoi(tokens[0])
	if err != nil {
		return 0, false
	}

	if len(tokens) == 1 {
		return result, true
	}

	if (len(tokens)-1)%2 != 0 {
		return 0, false // Requires pairs of operator+number
	}

	for i := 1; i < len(tokens); i += 2 {
		op := tokens[i]
		numStr := tokens[i+1]

		num, err := strconv.Atoi(numStr)
		if err != nil {
			return 0, false
		}

		switch op {
		case "+":
			result += num
		case "-":
			result -= num
		case "*":
			result *= num
		case "/":
			if num == 0 {
				return 0, false // Avoid division by zero
			}
			result /= num
		default:
			return 0, false // Invalid operator
		}
	}

	return result, true
}

File: src/2025-03-04-12-20-21--deepseek-vr-0304/python/exercises/practice/wordy/wordy.py
def answer(question):
    question = question.strip()
    if not question.startswith("What is ") or not question.endswith("?"):
        raise ValueError("syntax error")
    content = question[8:-1].strip()
    if not content:
        raise ValueError("syntax error")
    
    content = content.replace("multiplied by", "*").replace("divided by", "/")
    tokens = content.split()
    tokens = [t for t in tokens if t]
    
    if not tokens:
        raise ValueError("syntax error")
    
    # Check structure: alternating numbers and operations
    for i in range(len(tokens)):
        if i % 2 == 0:
            # Even index: must be a number
            try:
                int(tokens[i])
            except ValueError:
                raise ValueError("syntax error")
        else:
            # Odd index: must be a valid operation
            op = tokens[i]
            if op not in ('plus', 'minus', '*', '/'):
                # Check if the token is a number, which is invalid here
                try:
                    int(op)
                    # It's a number in an operation position → syntax error
                    raise ValueError("syntax error")
                except ValueError:
                    # Not a number → unknown operation
                    raise ValueError("unknown operation")
    
    # Now evaluate the operations
    result = int(tokens[0])
    operations = {
        "plus": lambda a, b: a + b,
        "minus": lambda a, b: a - b,
        "*": lambda a, b: a * b,
        "/": lambda a, b: a // b
    }
    
    for i in range(1, len(tokens), 2):
        op = tokens[i]
        num = int(tokens[i + 1])
        result = operations[op](result, num)
    
    return result
